gradle.properties

```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-ideaVersion = IC-15.0.6
+ideaVersion = IU-162.1447.7
```

Comment:
Indeed I did.  Even though the changes in this PR are backward-compatible, bug fixes and behavior changes to the 2016.2 IDEA platform meant that I had to make some changes to our tests, so that they pass in new versions (but necessarily fail when building against older versions).  

I committed this change because this code will not pass a CI build targeting the older versions of the IDEA platform, even though it should run, once built, without issue for those versions. (The change from jdk 7->8 in travis.yml follows from this - the newer IDEA platform needs java 8).